units/recalculate_change_points.go

```diff
@@ -42,200 +42,88 @@ func makeChangePointsJob() *recalculateChangePointsJob {
-	return bson.M{
-		"info.project":   grouping.Id.Project,
-		"info.variant":   grouping.Id.Variant,
-		"info.task_name": grouping.Id.Task,
-		"info.test_name": grouping.Id.Test,
+func makeMessage(msg string, id model.TimeSeriesId) message.Fields {
+	return message.Fields{
+		"message": msg,
+		"project": id.Id.Project,
+		"variant": id.Id.Variant,
+		"task":    id.Id.Task,
+		"test":    id.Id.Test,
 	}
-
-}
-
-func makeTimeSeriesPipeline(grouping MetricGrouping) []bson.M {
-	var pipeline []bson.M
-
-	pipeline = append(pipeline, bson.M{
-		"match": makeSeriesFilter(grouping),
-	})
-
-	pipeline = append(pipeline, bson.M{
-		"$match": bson.M{
-			"info.order": bson.M{
-				"$exists": true,
-			},
-			"info.mainline": true,
-		},
-	})
-
-	pipeline = append(pipeline, bson.M{
-		"$unwind": "$rollups.stats",
-	})
-
-	pipeline = append(pipeline, bson.M{
-		"$project": bson.M{
-			"measurement": "$rollups.stats.name",
-			"value":       "$rollups.stats.val",
-			"order":       "$info.order",
-		},
-	})
-
-	pipeline = append(pipeline, bson.M{
-		"$group": bson.M{
-			"_id": "$measurement",
-			"time_series": bson.M{
-				"$push": bson.M{
-					"id":    "$_id",
-					"value": "$value",
-					"order": "$order",
-				},
-			},
-		},
-	})
-
-	return pipeline
-}
-
-type errorMsg = map[string]interface{}
-
-type timeSeriesElement struct {
-	PerfResultID string `bson:"id"`
-	Value        float64
-	Order        int
-}
-
-type measurementTimeSeries struct {
-	Measurement string              `bson:"_id"`
-	TimeSeries  []timeSeriesElement `bson:"times_series"`
 }
 
 func (j *recalculateChangePointsJob) Run(ctx context.Context) {
 	defer j.MarkComplete()
-	if j.env == nil {
-		j.env = cedar.GetEnvironment()
-	}
 	conf := model.NewCedarConfig(j.env)
 	err := conf.Find()
 	if err != nil {
-		grip.Error(errorMsg{
+		message.WrapError(err, message.Fields{
 			"message": "Unable to get cedar configuration",
-			"error":   err,
 		})
+		return
+	}
+	if conf.Flags.DisableSignalProcessing == true {
+		grip.Info(makeMessage("signal processing is disabled, skipping processing", j.TimeSeriesId))
+		return
 	}
+	if j.env == nil {
+		j.env = cedar.GetEnvironment()
+	}
+	db := j.env.GetDB()
 	detector := perf.NewMicroServiceChangeDetector(conf.ChangeDetector.URI, conf.ChangeDetector.User, conf.ChangeDetector.Token)
-	collection := j.env.GetDB().Collection("perf_results")
-	timeSeriesAggregator := makeTimeSeriesPipeline(j.metricGrouping)
-	cur, err := collection.Aggregate(ctx, timeSeriesAggregator)
+	cur, err := model.GetTimeSeries(ctx, db, j.TimeSeriesId)
 	defer cur.Close(ctx)
 	if err != nil {
-		grip.Error(errorMsg{
-			"message": "Unable to aggregate time series",
-			"project": j.metricGrouping.Id.Project,
-			"variant": j.metricGrouping.Id.Variant,
-			"task":    j.metricGrouping.Id.Task,
-			"test":    j.metricGrouping.Id.Test,
-			"error":   err,
-		})
+		message.WrapError(err, makeMessage("Unable to aggregate time series", j.TimeSeriesId))
```

Comment:
this isn't logged, you still have to pass the message to a log statment (e.g. grip.Error()